File: scenario-1/js/whole_test.js
var webdriver = require('selenium-webdriver');

function wholeTest() {
    start = new Date().getTime();

    var driver = new webdriver.Builder().
        withCapabilities(webdriver.Capabilities.chrome()).
        build();

    driver.get('http://mediawiki119.wikia.com');
    driver.manage().window().maximize()
    var element = driver.findElement(webdriver.By.id("searchInput"))

    element.sendKeys("qa")
    element.sendKeys(webdriver.Key.RETURN)

    driver.wait(function (num) {
        return driver.isElementPresent(webdriver.By.xpath("//iframe[@title='VisualDNA Analytics']"));
    }, 20000);

    driver.findElement(webdriver.By.css("article h1 a")).click();

    driver.wait(function () {
        return driver.getTitle().then(function (title) {
            return title == "QA - Mediawiki 1.19 test Wiki";
        });
    }, 10000);

    driver.wait(function () {
        return driver.isElementPresent(webdriver.By.xpath("//iframe[@title='VisualDNA Analytics']"));
    }, 20000);

    var text = "ASDASD" + num;

    driver.findElement(webdriver.By.css("nav.contribute")).then(function (elem) {
        driver.actions().mouseMove(elem).mouseDown().mouseUp()
            .perform();
    });

    driver.findElement(webdriver.By.css("nav li a[data-id='edit']")).click();

    driver.wait(function () {
        return driver.isElementPresent(webdriver.By.css("div.cke_contents iframe"));
    }, 20000);

    driver.switchTo(driver.findElement(webdriver.By.css("div.cke_contents iframe")));
    driver.findElement(webdriver.By.css("body")).then(function (elem) {
        driver.actions().doubleClick(elem).sendKeys(webdriver.Key.BACK_SPACE)
            .sendKeys(text).perform();
    });
    driver.switchTo().defaultContent();
    driver.findElement(webdriver.By.css("input#wpSave")).click();

    driver.wait(function () {
        return driver.getTitle().then(function (title) {
            return title == "QA - Mediawiki 1.19 test Wiki";
        });
    }, 10000);

    driver.wait(function () {
        return driver.findElement(webdriver.By.css("#WikiaArticle p")).getText().then(function (pText) {
            return pText == text;
        });
    }, 5000);

    driver.quit().then(function () {
        console.log((new Date().getTime() - start) / 1000);
    });
}

var flow = webdriver.promise.controlFlow();

var num = 0;

for (var i = 0; i < 120; ++i) {
    flow.execute(wholeTest).then(function () {
        num += 1;
    })
}

File: scenario-1/java/src/com/company/WholeTest.java
package com.company;

import java.util.Date;

import org.openqa.selenium.By;
import org.openqa.selenium.Keys;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class WholeTest {

  public static void main(String args[]) {
    System.setProperty("webdriver.chrome.silentOutput", "true");
    System.setProperty("webdriver.chrome.verboseLogging", "true");
    System.setProperty("webdriver.chrome.logfile", "chromelog.log");

    for (int i = 0; i < 120; i++) {
      long start = new Date().getTime();

      WebDriver driver = new ChromeDriver(DesiredCapabilities.chrome());

      driver.get("http://www.mediawiki119.wikia.com");
      driver.manage().window().maximize();
      WebElement searchInput = driver.findElement(By.id("searchInput"));
      searchInput.sendKeys("qa");
      searchInput.findElement(By.id("searchInput")).sendKeys(Keys.RETURN);

      new WebDriverWait(driver, 20).until(ExpectedConditions.presenceOfElementLocated(By
          .xpath("//iframe[@title='VisualDNA Analytics']")));
      driver.findElement(By.cssSelector("article h1 a")).click();

      new WebDriverWait(driver, 10).until(ExpectedConditions
          .titleIs("QA - Mediawiki 1.19 test Wiki"));

      String text = "ASDASD" + i;

      WebElement contributeButton = driver.findElement(By.cssSelector("nav.contribute"));

      new WebDriverWait(driver, 10).until(ExpectedConditions.presenceOfElementLocated(By
          .xpath("//iframe[@title='VisualDNA Analytics']")));

      new Actions(driver).click(contributeButton)
          .moveToElement(contributeButton.findElement(By.cssSelector("li a[data-id='edit']")))
          .click().perform();

      new WebDriverWait(driver, 20).until(ExpectedConditions.visibilityOfElementLocated(By
          .cssSelector("div.cke_contents iframe")));
      driver.switchTo().frame(driver.findElement(By.cssSelector("div.cke_contents iframe")));
      driver.findElement(By.tagName("body")).clear();
      driver.findElement(By.tagName("body")).sendKeys(text);
      driver.switchTo().defaultContent();
      driver.findElement(By.cssSelector("input#wpSave")).click();

      new WebDriverWait(driver, 10).until(ExpectedConditions
          .titleIs("QA - Mediawiki 1.19 test Wiki"));

      assert driver.findElement(By.cssSelector("#WikiaArticle p")).getText().equals(text);

      driver.quit();
      System.out.println((new Date().getTime() - start) / 1000f);
    }
  }
}
